### PR TITLE
Fix/converter

### PR DESCRIPTION
**Pull request by ZelikSV**

### CODE REVIEW

File: src/api/axios/index.js [hunk 1]
```diff
@@ -1,6 +1,6 @@
 import axios from 'axios';
 
-export const getCurrencyApi = () =>
-    axios
-      .get('https://api.privatbank.ua/p24api/pubinfo?json&exchange&coursid=5')
-      .then(item => item.data);
\ No newline at end of file
+export default () =>
```
Comment (motifej) [inline src/api/axios/index.js]: it is'nt correct, why request to privatbank it is default export from api?

File: src/components/Currency/index.js [hunk 2]
```diff
@@ -1,175 +1,153 @@
...
-import FilledInput from '@material-ui/core/FilledInput';
-import TextField from '@material-ui/core/TextField';
+import { Field, reduxForm } from 'redux-form';
+import { compose } from 'recompose';
 import Button from '@material-ui/core/Button';
-import './style.css';
+import FormControl from '@material-ui/core/FormControl';
+import { withStyles } from '@material-ui/core/styles';
+
+import { feeConvert } from '../../mocks/db';
```
Comment (motifej) [inline src/components/Currency/index.js]: use absolute path

File: src/components/Currency/index.js [hunk 3]
```diff
@@ -1,175 +1,153 @@
...
-    super(props);
-    this.state = {
-      currencyBuy: 'EUR',
-      currencySell: 'USD',
-      amountSell: 0,
-      amountBuy: 0,
-      fee: 1,
-    };
+  componentDidMount() {
+    this.props.addCurrency();
```
Comment (motifej) [inline src/components/Currency/index.js]: it is not clear - why in did mount we call action - addCurrency, i guess you should rename it

File: src/components/Currency/index.js [hunk 4]
```diff
@@ -1,175 +1,153 @@
...
-    this.setState({ amountSell: amountBuy });
-    this.setState({ amountBuy: amountSell });
+    const { currenciesCount, change } = this.props;
+    const {
+      currencyBuy,
+      currencySell,
+      amountSell,
+      amountBuy,
+    } = currenciesCount.values;
+    change('currencyBuy', currencySell);
```
Comment (motifej) [inline src/components/Currency/index.js]: looks strange, 4 calls

File: src/components/Currency/index.js [hunk 5]
```diff
@@ -1,175 +1,153 @@
...
-    const {
-      currencyBuy,
-      currencySell,
-      fee,
-      amountBuy,
-      amountSell,
-    } = this.state;
-    const { handleChange, _changeCurrencies, handleInput, _buyCurrency } = this;
+    const { currencies, classes, handleSubmit } = this.props;
+    const { _buyCurrency, _changeCurrencies } = this;
```
Comment (motifej) [inline src/components/Currency/index.js]: it is unnecessarily, don't use destructure for one level

File: src/helpers/converter.helper.js [hunk 6]
```diff
@@ -0,0 +1,38 @@
+const convertToUa = (from, to) => {
+  const result = from * to;
```
Comment (motifej) [inline src/helpers/converter.helper.js]: explain why we need variable? just 
convertToUa = (from, to) => from * to;

File: src/helpers/converter.helper.js [hunk 7]
```diff
@@ -0,0 +1,38 @@
...
+
+  return result;
+};
+
+const convertFromUa = (from, to) => {
```
Comment (motifej) [inline src/helpers/converter.helper.js]: the same as above

File: src/pages/Converter/index.js [hunk 8]
```diff
@@ -1,24 +1,46 @@
...
-  render() {
-    const { currencies } = this.props;
-    return (
-      <div className="converter-wrapper">
-        <Currency currencies={currencies} />
-      </div>
-    );
+const Converter = ({currencies, addCurrency, buyConvertCurrency, currenciesCount, countCurrency, amountBuy }) => {
+
+  const submit = values => {
```
Comment (motifej) [inline src/pages/Converter/index.js]: it is not necessary

File: src/pages/Converter/index.js [hunk 9]
```diff
@@ -1,24 +1,46 @@
...
+      addCurrency={addCurrency}
+      countCurrency={countCurrency}
+      currenciesCount={currenciesCount}
+      amountBuy={amountBuy}
+    />
+  </div>);
+};
+
+
+const mapStateToProps = (state) => {
```
Comment (motifej) [inline src/pages/Converter/index.js]: use destructure

({ combineEvents, form }) => ({
currencies: combineEvents.currencies,
...
})

File: src/reducers/currencyReducer/index.js [hunk 10]
```diff
@@ -1,14 +1,26 @@
...
-    default:
-      return state;
-  }
-}
-
-export default currencyReducer;
+export default combineEvents(
+  {
+    [ADD_CURRENCY_ASYNC]: (state, action) => ({
+      ...state,
```
Comment (motifej) [inline src/reducers/currencyReducer/index.js]: it is not necessary, just 

[ADD_CURRENCY_ASYNC]: (state, { payload }) => ({ currencies: payload  }),

File: src/sagas/currencySaga/index.js [hunk 11]
```diff
@@ -1,13 +1,37 @@
...
-export default function* watchGetAllCurrencies() {
+function* countCurrencies(action) {
+  const {
+    currencies,
+    currencySell,
+    currencyBuy,
+    amountSell,
+    fee,
+  } = action.payload;
+  const buyValue = yield call(() =>
```
Comment (motifej) [inline src/sagas/currencySaga/index.js]: it is not correct, do like - call(buyCurrency, .....) and you should use yield for async function, not sync!!!

File: src/components/Currency/index.js (hunk 2 above)
Comment (ZelikSV) [reply src/components/Currency/index.js]: have done

File: src/components/Currency/index.js (hunk 5 above)
Comment (ZelikSV) [reply src/components/Currency/index.js]: Ok

File: src/helpers/converter.helper.js (hunk 6 above)
Comment (ZelikSV) [reply src/helpers/converter.helper.js]: cleaned

File: src/pages/Converter/index.js (hunk 9 above)
Comment (ZelikSV) [reply src/pages/Converter/index.js]: Ok

File: src/sagas/currencySaga/index.js (hunk 11 above)
Comment (ZelikSV) [reply src/sagas/currencySaga/index.js]: have changed

File: src/pages/Converter/index.js [hunk 12]
```diff
@@ -1,24 +1,46 @@
...
-    );
+const Converter = ({currencies, addCurrency, buyConvertCurrency, currenciesCount, countCurrency, amountBuy }) => {
+
+  const submit = values => {
+    buyConvertCurrency(values);
+  };
+
+  return (<div className="converter-wrapper">
+    <Currency
+      onSubmit={submit}
```
[the remark this answers is not in the document]
Comment (ZelikSV) [reply src/pages/Converter/index.js]: Ok

File: src/reducers/currencyReducer/index.js (hunk 10 above)
Comment (ZelikSV) [reply src/reducers/currencyReducer/index.js]: Without state it doesn't work

File: package.json [hunk 13]
```diff
@@ -19,6 +19,7 @@
     "prop-types": "^15.7.2",
     "react": "^16.8.2",
     "react-dom": "^16.8.2",
+    "react-hot-loader": "^4.8.0",
```
Comment (motifej) [inline package.json]: why did you add this module?

File: src/actions/currencyAction.js [hunk 14]
```diff
@@ -1,23 +1,38 @@
...
+export const buyConvertCurrency = value => ({
   type: BUY_CURRENCY,
-  payload: data,
+  payload: value,
 });
 
-export const chooseCurrencySell = data => ({
-  type: SELL_CURRENCY,
-  payload: data,
+export const countCurrency = (
```
Comment (motifej) [inline src/actions/currencyAction.js]: currencies,
 item }) => ({
  type: COUNT_CURRENCY,
  payload: {
    currencies,
    ...item
})

File: src/components/Currency/index.js [hunk 15]
```diff
@@ -1,175 +1,149 @@
...
+    const { currencies, classes, handleSubmit } = this.props;
+    const selectsOptions = oppositeCurrency =>
+      currencies
+        .filter(item => item.ccy !== oppositeCurrency)
+        .map(item => (
+          <option value={item.ccy} key={item.ccy}>
+            {item.ccy}
+          </option>
+        ));
+    const feeSelectsOption = () =>
```
Comment (motifej) [inline src/components/Currency/index.js]: move to render

File: src/components/Currency/index.js [hunk 16]
```diff
@@ -1,175 +1,149 @@
...
-      amountBuy,
-      amountSell,
-    } = this.state;
-    const { handleChange, _changeCurrencies, handleInput, _buyCurrency } = this;
+    const { currencies, classes, handleSubmit } = this.props;
+    const selectsOptions = oppositeCurrency =>
+      currencies
+        .filter(item => item.ccy !== oppositeCurrency)
+        .map(item => (
+          <option value={item.ccy} key={item.ccy}>
```
Comment (motifej) [inline src/components/Currency/index.js]: move to render

File: src/components/Currency/index.js [hunk 17]
```diff
@@ -1,175 +1,149 @@
...
         </Button>
-      </div>
+      </form>
     );
   }
 }
 
-export default Currency;
+export default compose(
+  reduxForm({
```
Comment (motifej) [inline src/components/Currency/index.js]: move abov like - const withForm = reduxForm({ ... })
and
export default compose(
  withForm,
  withStyles(styles),
)(Currency);

File: src/actions/currencyAction.js (hunk 14 above)
Comment (ZelikSV) [reply src/actions/currencyAction.js]: have done

File: src/reducers/currencyReducer/index.js (hunk 10 above)
Comment (AndrZakhara) [reply src/reducers/currencyReducer/index.js]: I solved it in this way:
function combineEvents() {
....
    if (action.type in reducers) {
      return { ...state, ...reducers[action.type](state, action) };
    }
    if ('default' in reducers) {
      return { ...state, ...reducers.default(state, action) };
    }
...
}

File: src/components/Currency/index.js [hunk 18]
```diff
@@ -1,175 +1,169 @@
...
-    const secondConvert = this.convertFromUa(firstConvert, currencies[a].sale);
-    const countWithTax = secondConvert - this.countTax(secondConvert, fee);
-
-    this.setState({ amountBuy: Math.trunc(countWithTax * 100) / 100 });
+  
+  sendCurrenciesDeal = e => {
+    const datas = moment().format('L');
+    console.log(datas);
+    console.log(this.props.values);
+    e.preventDefault();
```
Comment (motifej) [inline src/components/Currency/index.js]: always a first line

File: src/components/Currency/index.js [hunk 19]
```diff
@@ -1,175 +1,169 @@
...
-    );
-    const secondConvert = this.convertFromUa(firstConvert, currencies[a].sale);
-    const countWithTax = secondConvert - this.countTax(secondConvert, fee);
-
-    this.setState({ amountBuy: Math.trunc(countWithTax * 100) / 100 });
+  
+  sendCurrenciesDeal = e => {
+    const datas = moment().format('L');
+    console.log(datas);
+    console.log(this.props.values);
```
Comment (motifej) [inline src/components/Currency/index.js]: remove

File: src/components/Currency/index.js [hunk 20]
```diff
@@ -1,175 +1,169 @@
...
+          <FormControl className={classes.formControl}>
+            <p className={classes.text}>give back</p>
+            <Field
+              className={classes.select}
+              name="currencySell"
+              component="select"
+              onChange={this.buyCurrency}>
+              {
+                currencies
+                  .filter(item => item.ccy !== values.currencyBuy)
```
Comment (motifej) [inline src/components/Currency/index.js]: avoid any logic in jsx

File: src/components/Currency/index.js [hunk 21]
```diff
@@ -1,175 +1,169 @@
...
+          <FormControl className={classes.formControl}>
+            <p className={classes.text}>we get</p>
+            <Field
+              className={classes.select}
+              name="currencyBuy"
+              component="select"
+              onChange={this.buyCurrency}>
+              {
+                currencies
+                  .filter(item => item.ccy !== values.currencySell)
```
Comment (motifej) [inline src/components/Currency/index.js]: avoid any logic in jsx

File: src/helpers/converter.helper.js [hunk 22]
```diff
@@ -0,0 +1,24 @@
...
+  const firstConvert = convertToUa(AmountSell, array[indexSellCurrency].buy);
+  const secondConvert = convertFromUa(
+    firstConvert,
+    array[indexBuyCurrency].sale,
+  );
+  const countWithTax = secondConvert - countTax(secondConvert, tax);
+
+  return Math.trunc(countWithTax * 100) / 100;
+};
+export const swappingVariables = (currencyBuy, currencySell, amountSell, amountBuy, cb) => {
```
Comment (motifej) [inline src/helpers/converter.helper.js]: remember me about this code

File: src/utils/combineEvents.js [hunk 23]
```diff
@@ -0,0 +1,14 @@
+export default function combineEvents(reducers, initialState) {
```
Comment (motifej) [inline src/utils/combineEvents.js]: one person push - the other pull
